File: Krutarth_24B0055/Week3/10_Koko_Eating_Bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1;
        int r = *max_element(piles.begin(), piles.end());

        while (l < r){
            int mid = l + (r-l) / 2;
            int hours = 0;

            for(int bananas : piles){
                hours += (bananas + mid - 1) / mid;
            }

            if (hours <= h){
                r = mid;
            }
            else{
                l = mid + 1;
            }
        }

        return l;
    }
};

File: Akshit_22b2531/week3/10_koko_eating_bananas.py
def min_eating_speed(piles, h):
    left, right = 1, max(piles)
    while left < right:
        mid = (left + right) // 2
        hours = sum((pile + mid - 1) // mid for pile in piles)
        if hours > h:
            left = mid + 1
        else:
            right = mid
    return left

# Example usage:
piles = [3,6,7,11]
h = 8
print(min_eating_speed(piles, h))  # Output: 4 